{"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/init(errorDescription:recoverySuggestion:error:)"},"hierarchy":{"paths":[["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/AmplifyError-Implementations"]]},"schemaVersion":{"patch":0,"minor":3,"major":0},"sections":[],"abstract":[{"type":"text","text":"AmplifyErrors must be able to be initialized from an underlying error. If an AmplifyError is created"},{"text":" ","type":"text"},{"type":"text","text":"with this initializer, it must store the underlying error in the "},{"code":"underlyingError","type":"codeVoice"},{"type":"text","text":" property so it can be"},{"text":" ","type":"text"},{"text":"inspected later.","type":"text"}],"primaryContentSections":[{"declarations":[{"platforms":["iOS"],"tokens":[{"kind":"keyword","text":"init"},{"kind":"text","text":"("},{"text":"errorDescription","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier","text":"ErrorDescription","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/ErrorDescription"},{"kind":"text","text":" = \"An unknown error occurred\", "},{"text":"recoverySuggestion","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"RecoverySuggestion","kind":"typeIdentifier","preciseIdentifier":"s:7Amplify18RecoverySuggestiona","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/RecoverySuggestion"},{"kind":"text","text":" = \"(Ignored)\", "},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"},{"kind":"text","text":")"}],"languages":["swift"]}],"kind":"declarations"},{"kind":"content","content":[{"level":2,"type":"heading","text":"Discussion","anchor":"discussion"},{"type":"paragraph","inlineContent":[{"type":"text","text":"Implementations of this method should handle the case where "},{"type":"codeVoice","code":"error"},{"type":"text","text":" is already an instance of "},{"type":"codeVoice","code":"Self"},{"type":"text","text":", and simply"},{"type":"text","text":" "},{"type":"text","text":"return "},{"type":"codeVoice","code":"self"},{"type":"text","text":" as the incoming "},{"type":"codeVoice","code":"error"},{"text":".","type":"text"}]}]}],"kind":"symbol","metadata":{"fragments":[{"kind":"identifier","text":"init"},{"text":"(","kind":"text"},{"text":"errorDescription","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"ErrorDescription","preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"recoverySuggestion","kind":"externalParam"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:7Amplify18RecoverySuggestiona","text":"RecoverySuggestion"},{"text":", ","kind":"text"},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"},{"kind":"text","text":")"}],"modules":[{"name":"Amplify"}],"role":"symbol","title":"init(errorDescription:recoverySuggestion:error:)","roleHeading":"Initializer","symbolKind":"init","externalID":"s:7Amplify9AuthErrorO16errorDescription18recoverySuggestion0D0ACSS_SSs0C0_ptcfc","extendedModule":"Amplify"},"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/amplify\/autherror\/init(errordescription:recoverysuggestion:error:)"]}],"references":{"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/init(errorDescription:recoverySuggestion:error:)":{"url":"\/documentation\/amplify\/autherror\/init(errordescription:recoverysuggestion:error:)","fragments":[{"kind":"identifier","text":"init"},{"text":"(","kind":"text"},{"text":"errorDescription","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier","text":"ErrorDescription"},{"text":", ","kind":"text"},{"text":"recoverySuggestion","kind":"externalParam"},{"kind":"text","text":": "},{"preciseIdentifier":"s:7Amplify18RecoverySuggestiona","kind":"typeIdentifier","text":"RecoverySuggestion"},{"kind":"text","text":", "},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"text":"Error","kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":")"}],"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/init(errorDescription:recoverySuggestion:error:)","abstract":[{"type":"text","text":"AmplifyErrors must be able to be initialized from an underlying error. If an AmplifyError is created"},{"text":" ","type":"text"},{"text":"with this initializer, it must store the underlying error in the ","type":"text"},{"code":"underlyingError","type":"codeVoice"},{"type":"text","text":" property so it can be"},{"text":" ","type":"text"},{"text":"inspected later.","type":"text"}],"role":"symbol","title":"init(errorDescription:recoverySuggestion:error:)","type":"topic","kind":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify/RecoverySuggestion":{"url":"\/documentation\/amplify\/recoverysuggestion","title":"RecoverySuggestion","type":"topic","abstract":[{"type":"text","text":"See: "},{"identifier":"x-source-tag:\/\/AmplifyError.recoverySuggestion","type":"reference","isActive":true}],"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/RecoverySuggestion","fragments":[{"text":"typealias","kind":"keyword"},{"text":" ","kind":"text"},{"text":"RecoverySuggestion","kind":"identifier"}],"role":"symbol","kind":"symbol","navigatorTitle":[{"text":"RecoverySuggestion","kind":"identifier"}]},"doc://amplify-swift.Amplify/documentation/Amplify/ErrorDescription":{"url":"\/documentation\/amplify\/errordescription","title":"ErrorDescription","type":"topic","abstract":[{"type":"text","text":"See: "},{"type":"reference","identifier":"x-source-tag:\/\/AmplifyError.errorDescription","isActive":true}],"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/ErrorDescription","fragments":[{"kind":"keyword","text":"typealias"},{"text":" ","kind":"text"},{"kind":"identifier","text":"ErrorDescription"}],"role":"symbol","kind":"symbol","navigatorTitle":[{"kind":"identifier","text":"ErrorDescription"}]},"doc://amplify-swift.Amplify/documentation/Amplify":{"url":"\/documentation\/amplify","type":"topic","title":"Amplify","role":"collection","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","kind":"symbol","abstract":[]},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError":{"navigatorTitle":[{"kind":"identifier","text":"AuthError"}],"kind":"symbol","role":"symbol","url":"\/documentation\/amplify\/autherror","title":"AuthError","type":"topic","abstract":[{"type":"text","text":"Amplify error raised in the Auth category."}],"fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"text":"AuthError","kind":"identifier"}],"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError"},"x-source-tag://AmplifyError.recoverySuggestion":{"type":"link","identifier":"x-source-tag:\/\/AmplifyError.recoverySuggestion","titleInlineContent":[{"text":"AmplifyError.recoverySuggestion","type":"text"}],"title":"AmplifyError.recoverySuggestion","url":"x-source-tag:\/\/AmplifyError.recoverySuggestion"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/AmplifyError-Implementations":{"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/AmplifyError-Implementations","abstract":[],"title":"AmplifyError Implementations","kind":"article","role":"collectionGroup","type":"topic","url":"\/documentation\/amplify\/autherror\/amplifyerror-implementations"},"x-source-tag://AmplifyError.errorDescription":{"type":"link","identifier":"x-source-tag:\/\/AmplifyError.errorDescription","titleInlineContent":[{"type":"text","text":"AmplifyError.errorDescription"}],"title":"AmplifyError.errorDescription","url":"x-source-tag:\/\/AmplifyError.errorDescription"}}}